docs/source/user_guide/server_types.rst

```diff
@@ -0,0 +1,146 @@
+.. _user_guide_server_types:
+
+===========================
+Client-server communication
+===========================
+
+Terminology
+-----------
+
+DPF is based on a **client-server** architecture. It allows **local** or **remote**
```

Comment:
In the getting started section, I went for communication "in process" or "through the network" to be closer to the truth. We can use either one or the other but should be consistent maybe

Reply:
@PProfizi I took that comment into account, and the discussion we had yesterday; we can discuss it if needed.